core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/DetailedSampleImpl.java

```diff
@@ -159,6 +160,15 @@ public void setGroupDescription(String groupDescription) {
     this.groupDescription = nullifyStringIfBlank(groupDescription);
   }
 
+  @Override
+  public Optional<DetailedSample> getEffectiveGroupIdSample() {
+    for (DetailedSample sample = this; sample != null; sample = sample.getParent()) {
+      String groupId = sample.getGroupId();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "groupId" which hides the field declared at line 59. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)